hedera-node/src/test/java/com/hedera/services/tokens/HederaTokenStoreTest.java

```diff
@@ -959,8 +963,62 @@ public void updateRejectsBadFreezeKey() {
 		assertEquals(INVALID_FREEZE_KEY, outcome);
 	}
 
+	@Test
+	public void treasuryRemovalForTokenRemovesKeyWhenEmpty() {
+		Set<TokenID> tokenSet = new HashSet<>(Arrays.asList(misc));
+		subject.knownTreasuries.put(treasury, tokenSet);
+
+		subject.removeKnownTreasuryForToken(treasury, misc);
+
+		// expect:
+		assertFalse(subject.knownTreasuries.containsKey(treasury));
+		assertTrue(subject.knownTreasuries.isEmpty());
+	}
+
+	@Test
+	public void addKnownTreasuryWorks() {
+		Set<TokenID> tokenSet = new HashSet<>(Arrays.asList(misc));
+		subject.knownTreasuries.put(treasury, tokenSet);
+
+		// expect:
+		assertTrue(subject.knownTreasuries.containsKey(treasury));
+	}
+
+	@Test
+	public void removeKnownTreasuryWorks() {
+		Set<TokenID> tokenSet = new HashSet<>(Arrays.asList(misc, anotherMisc));
+		subject.knownTreasuries.put(treasury, tokenSet);
+
+		subject.removeKnownTreasuryForToken(treasury, misc);
+
+		// expect:
+		assertTrue(subject.knownTreasuries.containsKey(treasury));
+		assertEquals(1, subject.knownTreasuries.size());
+		assertTrue(subject.knownTreasuries.get(treasury).contains(anotherMisc));
+	}
+
+	@Test
+	public void isKnownTreasuryWorks() {
+		Set<TokenID> tokenSet = new HashSet<>(Arrays.asList(misc));
+
+		subject.knownTreasuries.put(treasury, tokenSet);
+
+		// expect:
+		assertTrue(subject.knownTreasuries.containsKey(treasury));
```

Comment:
In order for you to test the `isKnownTreasury` you need to actually call it. In the current unit test setup, you are testing the Java Map functionality